termwiz/src/cell.rs

```diff
@@ -881,17 +882,40 @@ impl Cell {
     }
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, FromDynamic, ToDynamic)]
+pub struct CellWidth {
+    pub first: u32,
+    pub last: u32,
+    pub width: u8,
+}
```

Comment:
From a layering perspective, I think this struct should move to the `config` crate, and that the `setcellwidths` function should move along with it.  I would probably also move `setcellwidths` to be an associated function of `CellWidth`

```rust
impl CellWidth {
   fn compile_to_map(widths: Option<Vec<Self>>) -> Option<Arc<HashMap<u32,u8>>> {
     ...
   }
}
```